backend/FwLite/FwDataMiniLcmBridge/Api/FwDataMiniLcmApi.cs

```diff
@@ -956,35 +955,43 @@ internal void RemoveComplexFormComponent(ILexEntry lexEntry, ComplexFormComponen
             lexComponent = entry;
         }
 
-        var entryRef = lexEntry.ComplexFormEntryRefs.Single();
-        if (!entryRef.ComponentLexemesRS.Remove(lexComponent))
+        foreach (var entryRef in lexEntry.ComplexFormEntryRefs)
         {
-            throw new InvalidOperationException("Complex form component not found, searched for " + lexComponent.ObjectIdName.Text);
+            if (entryRef.ComponentLexemesRS.Remove(lexComponent)) return;
         }
+        //not throwing to match CRDT behavior
     }
 
     internal void AddComplexFormType(ILexEntry lexEntry, Guid complexFormTypeId)
     {
-        ILexEntryRef? entryRef = lexEntry.ComplexFormEntryRefs.SingleOrDefault();
+        //do the same thing as LCM, use the first when adding if there's more than one
+        ILexEntryRef? entryRef = lexEntry.ComplexFormEntryRefs.FirstOrDefault();
         if (entryRef is null)
         {
-            entryRef = Cache.ServiceLocator.GetInstance<ILexEntryRefFactory>().Create();
-            lexEntry.EntryRefsOS.Add(entryRef);
-            entryRef.RefType = LexEntryRefTags.krtComplexForm;
-            entryRef.HideMinorEntry = 0;
+            entryRef = AddComplexFormLexEntryRef(lexEntry);
         }
 
         var lexEntryType = ComplexFormTypesFlattened.Single(c => c.Guid == complexFormTypeId);
         entryRef.ComplexEntryTypesRS.Add(lexEntryType);
     }
 
+    internal ILexEntryRef AddComplexFormLexEntryRef(ILexEntry lexEntry)
+    {
+        var entryRef = Cache.ServiceLocator.GetInstance<ILexEntryRefFactory>().Create();
+        lexEntry.EntryRefsOS.Add(entryRef);
+        entryRef.RefType = LexEntryRefTags.krtComplexForm;
+        entryRef.HideMinorEntry = 0;
+        return entryRef;
+    }
+
     internal void RemoveComplexFormType(ILexEntry lexEntry, Guid complexFormTypeId)
     {
-        ILexEntryRef? entryRef = lexEntry.ComplexFormEntryRefs.SingleOrDefault();
-        if (entryRef is null) return;
-        var lexEntryType = entryRef.ComplexEntryTypesRS.SingleOrDefault(c => c.Guid == complexFormTypeId);
-        if (lexEntryType is null) return;
-        entryRef.ComplexEntryTypesRS.Remove(lexEntryType);
+        foreach (var entryRef in lexEntry.ComplexFormEntryRefs)
+        {
+            var lexEntryType = entryRef.ComplexEntryTypesRS.SingleOrDefault(c => c.Guid == complexFormTypeId);
+            if (lexEntryType is null) continue;
+            entryRef.ComplexEntryTypesRS.Remove(lexEntryType);
```

Comment:
In case of duplicates, we only want to delete one occurence.
```suggestion
            if (entryRef.ComplexEntryTypesRS.Remove(lexEntryType)) break;
```